File: paper_artifacts/experiments/llr8/artifacts/sources/kimi27sglang-fortran/tsvc_2_vpvts__submitted.f90
subroutine tsvc_2_vpvts_fp64(a, b, LEN_1D, S) bind(c, name='tsvc_2_vpvts_fp64')
    use iso_c_binding, only: c_int64_t, c_double
    implicit none
    integer(c_int64_t), value, intent(in) :: LEN_1D, S
    real(c_double), intent(inout) :: a(LEN_1D)
    real(c_double), intent(in) :: b(LEN_1D)
    integer(c_int64_t) :: i
    real(c_double) :: ds

    ds = real(S, c_double)
    !$omp parallel do simd schedule(static) if(len_1d > 1000)
    do i = 1, LEN_1D
        a(i) = a(i) + b(i) * ds
    end do
    !$omp end parallel do simd
end subroutine tsvc_2_vpvts_fp64

File: paper_artifacts/experiments/llr8/artifacts/sources/kimi27sglang-c-skills/tsvc_2_vpvts__submitted.c
#include <stdint.h>
#include <omp.h>

void tsvc_2_vpvts_fp64(double *restrict a, const double *restrict b, const int64_t LEN_1D, const int64_t S) {
    const double s = (double)S;
    #pragma omp parallel for simd schedule(static)
    for (int64_t i = 0; i < LEN_1D; ++i) {
        a[i] += b[i] * s;
    }
}
